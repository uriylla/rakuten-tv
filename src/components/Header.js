import React from 'react'

export default class Header extends React.Component {
  render () {
    const { movie } = this.props;
    return (
    <div className="header">
      <h1>{movie || 'Rakuten TV'}</h1>
    </div>
    )
  }
}