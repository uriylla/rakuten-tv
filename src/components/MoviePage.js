import React from 'react'
import { connect } from 'react-redux'
import {Â getSelectedMovie } from '../selectors/movies'
import { withRouter } from 'react-router'

export class MoviePage extends React.Component {

  constructor (props) {
    super(props)
    this.goToTrailer = this.goToTrailer.bind(this)
  }

  goToTrailer () {
    const { history, id } = this.props;
    history.push(`/movie/${id}/trailer`) 
  }

  render () {
    const { movie } = this.props
    if(!movie) return <div className="movie">Loading...</div>
    var sectionStyle = { backgroundImage: `url(${this.props.movie.images.snapshot})`};
    return (
      <div className="movie">
        <div className="movie__cover"  style={ sectionStyle }>
          <div className="cover-gradient"></div>
          <div className="movie__trailer-btn" onClick={this.goToTrailer}>
            <span className="material-icons">play_circle_filled</span>
            <p>TRAILER</p>
          </div>
        </div>
        <div className="movie__info"></div>
      </div>
    )
  }
}

export default withRouter(MoviePage);