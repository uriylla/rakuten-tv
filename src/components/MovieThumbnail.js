import React from 'react';

export default class MovieThumbnail extends React.Component {
  render () {
    const { artwork_url, score } = this.props
    return (
      <div className="movie">
        <img className="movie__image" src={artwork_url}/>
      </div>
    )
  }
}