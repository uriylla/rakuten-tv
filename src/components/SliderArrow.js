import React from 'react'

export class SliderArrowPrev extends React.Component {
  render () {
    const { className, style, onClick } = this.props;
    const styles = {
      display: "flex",
      background: "linear-gradient(to right,rgba(23,23,23,.95) 0,transparent 100%)",
      height: "102%",
      padding: "0px 20px 0px 20px",
      width: "4%",
      left: "0px",
      zIndex: "2",
      justifyContent: "flex-start",
      alignItems: "center"
    }
    return (
      <div
        className={className}
        style={{...styles}}
        onClick={onClick}
      />
    )
  }
}

export class SliderArrowNext extends React.Component {
  render () {
    const { className, style, onClick } = this.props;
    const styles = {
      display: "flex",
      background: "linear-gradient(to left,rgba(23,23,23,.95) 0,transparent 100%)",
      height: "102%",
      padding: "0px 20px 0px 20px",
      width: "4%",
      right: "0px",
      zIndex: "2",
      justifyContent: "flex-end",
      alignItems: "center"
    }
    return (
      <div
        className={className}
        style={{...styles}}
        onClick={onClick}
      />
    )
  }
}