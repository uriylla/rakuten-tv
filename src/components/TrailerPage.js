import React from 'react'
import { withRouter } from 'react-router'

export class TrailerPage extends React.Component {

  constructor(props) {
    super(props)
    this.goBack = this.goBack.bind(this)
  }

  componentDidMount() {
    this.player = videojs('trailer-player', {
      autoplay: true,
      children: [
        'bigPlayButton',
        'controlBar'
      ],
      controls: true
    })
  }

  componentDidUpdate () {
    const { trailer } = this.props
    trailer && this.loadVideo(trailer)
  }

  componentWillUnmount () {
    this.player.dispose()
  }

  goBack () {
    const { history, id } = this.props
    history.push(`/movie/${id}`)
  }

  loadVideo (videoUrl) {
    this.player.src({
      src: videoUrl,
      type: 'application/dash+xml'
    });
    this.player.play();
  }

  render () {
    const { trailer } = this.props
    return (
      <div className="trailer-page">
        <span className="trailer-page__back material-icons" onClick={this.goBack} >arrow_back_ios</span>
        <video className="video-js vjs-big-play-centered" id="trailer-player"></video>
      </div>
    )
  }
}

export default withRouter(TrailerPage)