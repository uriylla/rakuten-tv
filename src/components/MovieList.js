import React from 'react'
import MovieThumbnail from './MovieThumbnail'
import {SliderArrowNext, SliderArrowPrev} from './SliderArrow'
import { connect } from 'react-redux'
import { selectListById } from '../selectors/movies'
import Slider from "react-slick";

export class MovieList extends React.Component {

  state = { activeSlide: 0 }

  render () {
    const current = this.state.activeSlide
    const total = this.props.movies.length
    const prevActive = current > 0
    const nextActive =  current < total - 4
    const settings = {
      draggable: false,
      infinite: false,
      speed: 1000,
      prevArrow: prevActive ? <SliderArrowPrev /> : undefined,
      nextArrow: <SliderArrowNext />,
      slidesToShow: 7.3,
      slidesToScroll: 5,
      afterChange: (index) => this.setState({activeSlide: index}),
      responsive: [
        {
          breakpoint: 1400,
          settings: {
            slidesToShow: 6.5,
            slidesToScroll: 4
          }
        },
        {
          breakpoint: 1140,
          settings: {
            slidesToShow: 5.5,
            slidesToScroll: 3
          }
        },
        {
          breakpoint: 910,
          settings: {
            slidesToShow: 4.5,
            slidesToScroll: 3
          }
        },
        {
          breakpoint: 730,
          settings: {
            slidesToShow: 3.5,
            slidesToScroll: 2
          }
        },
        {
          breakpoint: 560,
          settings: {
            slidesToShow: 2.5,
            slidesToScroll: 1
          }
        }
      ]
    }
    const { name, movies } = this.props
    return (
      <div className="movie-list">
        <h1 className="movie-list__title">{name}</h1>
        <Slider {...settings} >
          {movies.map( movie => <MovieThumbnail key={movie.id} {...movie} /> )}
        </Slider>
      </div>
    )
  }
}

const mapStateToProps = (state, {id}) => ({ ...selectListById(state.lists, id) })

export default connect(mapStateToProps)(MovieList)