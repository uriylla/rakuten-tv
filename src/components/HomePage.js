import React from 'react'
import MovieList from './MovieList'
import { connect } from 'react-redux'
import { selectLists } from '../selectors/movies'

export class HomePage extends React.Component {
  render () {
    const { lists } = this.props
    return (
      <div className="home">
        {lists && lists.map(({id}) => <MovieList key={id} id={id} /> )}
      </div>
    )
  }
}

const mapStateToProps = (state, {name}) => ({
  lists: selectLists(state.lists)
})

export default connect(mapStateToProps)(HomePage)