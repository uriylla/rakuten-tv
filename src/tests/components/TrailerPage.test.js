import React from 'react'
import { shallow, mount } from 'enzyme'
import { TrailerPage } from '../../components/TrailerPage'

it('should render TrailerPage correctly', () => {
  const wrapper = shallow(<TrailerPage />)
  expect(wrapper).toMatchSnapshot()
})

it('should render go to MoviePage on back button click', () => {
  const id = 'matrix'
  const videojs = () => {}
  const pushSpy = jest.fn()
  const wrapper = shallow(<TrailerPage id={id} history={{push: pushSpy}}/>)
  wrapper.find('.trailer-page__back').simulate('click')
  expect(pushSpy).toHaveBeenCalledWith(`/movie/${id}`)
})