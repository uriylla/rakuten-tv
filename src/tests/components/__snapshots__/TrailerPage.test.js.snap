// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TrailerPage correctly 1`] = `ShallowWrapper {}`;
